src/Recruit.Vacancies.Client/Infrastructure/QueryStore/QueryStore.cs

```diff
@@ -1,32 +1,46 @@
-using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Esfa.Recruit.Vacancies.Client.Domain.Projections;
 using Esfa.Recruit.Vacancies.Client.Domain.QueryStore;
 using Esfa.Recruit.Vacancies.Client.Infrastructure.Mongo;
 using Microsoft.Extensions.Options;
 using MongoDB.Driver;
-using System.Collections.Generic;
 using System.Threading.Tasks;
 
 namespace Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore
 {
     internal sealed class QueryStore : MongoDbCollectionBase, IQueryStoreReader, IQueryStoreWriter
     {
         private const string Database = "recruit";
-        private const string Collection = "vacancies";
+        private const string Collection = "queryViews";
 
         public QueryStore(IOptions<MongoDbConnectionDetails> details)
             : base(Database, Collection, details)
         {
         }
 
-        public async Task<IEnumerable<Vacancy>> GetVacanciesAsync(string employerAccountId)
+        public async Task<Dashboard> GetDashboardAsync(string key)
         {
-            var filter = Builders<Vacancy>.Filter.Eq(v => v.EmployerAccountId, employerAccountId);
-            filter = filter & Builders<Vacancy>.Filter.Eq(v => v.IsDeleted, false);
+            var filter = Builders<Dashboard>.Filter.Eq(d => d.ViewKey, key);
 
-            var collection = GetCollection<Vacancy>();
+            var collection = GetCollection<Dashboard>();
             var result = await collection.FindAsync(filter);
 
-            return result.ToEnumerable();
+            return result?.FirstOrDefault();
+        }
+
+        public async Task UpdateDashboardAsync(string key, Dashboard dashboard)
```

Comment:
Can we use 'key' as our 'id' in Mongo? Can we use a custom ObjectId so we can use EmployerAccountId as the id?

Reply:
you can just add the bsonid attribute or configure in the bson class mapper if you want a custom id OR just call the field "id"